script/tool/lib/src/analyze_command.dart

```diff
@@ -102,16 +102,25 @@ class AnalyzeCommand extends PackageLoopingCommand {
 
   @override
   Future<PackageResult> runForPackage(RepositoryPackage package) async {
-    // For non-example packages, fetch dependencies. 'flutter packages get'
-    // automatically runs 'pub get' in examples as part of handling the parent
-    // directory, which is guaranteed to come first in the package enumeration.
-    if (package.directory.basename != 'example' ||
-        !RepositoryPackage(package.directory.parent).pubspecFile.existsSync()) {
-      final int exitCode = await processRunner.runAndStream(
-          flutterCommand, <String>['packages', 'get'],
-          workingDir: package.directory);
-      if (exitCode != 0) {
-        return PackageResult.fail(<String>['Unable to get dependencies']);
+    // Analysis runs over the package and all subpackages, so all of them need
+    // `flutter packages get` run before analyzing. `example` packages can be
+    // skipped since 'flutter packages get' automatically runs `pub get` in
+    // examples as part of handling the parent directory.
+    final List<RepositoryPackage> packagesToGet = <RepositoryPackage>[
+      package,
+      ...await getSubpackages(package).toList(),
+    ];
+    for (final RepositoryPackage packageToGet in packagesToGet) {
+      if (packageToGet.directory.basename != 'example' ||
+          !RepositoryPackage(packageToGet.directory.parent)
+              .pubspecFile
+              .existsSync()) {
+        final int exitCode = await processRunner.runAndStream(
+            flutterCommand, <String>['packages', 'get'],
```

Comment:
This is extremely nitpicky, I'm sure I've not noticed it in dozens of other PRs!, but `flutter packages` is an undocumented alias (kept for backwards compatibility, IIRC) of: `flutter pub` 🤓

```
$ flutter --help | grep packages
  pub               Commands for managing Flutter packages.
```

(It doesn't show up even in `-v` mode of help!, but there's a `--packages` that has a different purpose.)

Reply:
(`flutter packages get` will probably work forever, but I'm flagging it just in case 😛)